client/src/types/general/Dot.ts

```diff
@@ -2,4 +2,11 @@
 export type Dot = {
     x: number;
     y: number;
-};
\ No newline at end of file
+};
+
+const ORIGIN_POINT: Dot = {
+    x: 0,
+    y: 0,
+};
+
+export default ORIGIN_POINT;
```

Comment:
I don't like this "origin" thing. I don't know why exactly you need it. I put separate review comments in every place it is used. If you agree with all of them and remove ORIGIN_POINT usage from there, be sure to remove it from here, too.